src/dune_rules/jsoo/jsoo_rules.ml

```diff
@@ -234,12 +254,21 @@ let exe_rule cc ~javascript_files ~src ~target ~flags =
       [ Resolve.Memo.args
           (let open Resolve.Memo.O in
            let+ libs = libs in
-           Command.Args.Deps (jsoo_runtime_files libs))
-      ; Deps (List.map ~f:Path.build javascript_files)
+           Command.Args.Deps
+             ((match ctarget with
+              | JS -> []
+              | Wasm -> wasm_runtime_files libs)
+             @ jsoo_runtime_files libs))
```

Comment:
does the wasm target depend on the jsoo_runtime_files ?

Reply:
The JavaScript runtime files are parsed to get purity information about primitives. We want also to include JavaScript files with no annotation, like `lib/virtualdom.compiled.js` in package `virtual_dom`. And I have just added a way to link JavaScript code referenced by the Wasm runtime code.

Reply:
@vouillon, do you plan to support sourcemap for the wasm target ? What's the content of the js file when target is wasm ?

Reply:
It would be nice to support them eventually. Probably not a high priority, though.
What are the issues with source maps?
The contents of the js file is currently a mix of js code generated by the js_of_ocaml linker and of javascript code directly copied from a file. I should certainly improve on this, and then one could add a sourcemap to this file.

Reply:
The issue with source map is that it might need to access sources to embed them. We do not currently track this kind of dependencies. Preventing us from using sandboxing.